camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+      CamusHourlySweeper.this.metrics.runnerStartTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+
+      job.getConfiguration().setBoolean(MAPRED_COMPRESS_MAP_OUTPUT, DEFAULT_MAPRED_COMPRESS_MAP_OUTPUT);
+
+      ((CamusSweeperJob) Class.forName(props.getProperty(CAMUS_SWEEPER_IO_CONFIGURER_CLASS)).newInstance()).setLogger(
+          LOG).configureJob(topicName, job);
+
+      setNumOfReducersAndSplitSizes();
+      submitMrJob();
+      moveTmpPathToOutputPath();
+
+      CamusHourlySweeper.this.metrics.mrFinishTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+    }
+
+    private void moveTmpPathToOutputPath() throws IOException {
+      Path oldPath = null;
+      if (fs.exists(outputPath)) {
+        oldPath = new Path("/tmp", "_old_" + job.getJobID());
+        moveExistingContentInOutputPathToOldPath(oldPath);
+      }
+
+      LOG.info("Moving " + tmpPath + " to " + outputPath);
+      mkdirs(fs, outputPath.getParent(), perm);
+
+      if (!fs.rename(tmpPath, outputPath)) {
+        fs.rename(oldPath, outputPath);
+        fs.delete(tmpPath, true);
+        throw new RuntimeException("Error: cannot rename " + tmpPath + " to " + outputPath);
+      }
+      deleteOldPath(oldPath);
+    }
+
+    private void deleteOldPath(Path oldPath) throws IOException {
+      if (oldPath != null && fs.exists(oldPath)) {
+        LOG.info("Deleting " + oldPath);
+        fs.delete(oldPath, true);
+      }
+    }
+
+    private void moveExistingContentInOutputPathToOldPath(Path oldPath) throws IOException {
+      LOG.info("Path " + outputPath + " exists. Overwriting.");
+      if (!fs.rename(outputPath, oldPath)) {
+        fs.delete(tmpPath, true);
+        throw new RuntimeException("Error: cannot rename " + outputPath + " to " + oldPath);
+      }
+    }
+
+    private void setNumOfReducersAndSplitSizes() throws IOException {
+      long inputSize = getInputSize();
+
+      int maxFiles = job.getConfiguration().getInt(MAX_FILES, DEFAULT_MAX_FILES);
+      int numTasks = Math.min((int) (inputSize / targetFileSize) + 1, maxFiles);
+
+      if (job.getNumReduceTasks() != 0) {
+        determineAndSetNumOfReducers(numTasks);
+      } else {
+        setSplitSizes(inputSize / numTasks);
+      }
+    }
+
+    private void submitMrJob() throws IOException, InterruptedException, ClassNotFoundException {
+      CamusHourlySweeper.this.metrics.mrSubmitTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+      job.submit();
+      runningJobs.add(job);
+
+      CamusHourlySweeper.this.metrics.mrStartRunningTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+
+      LOG.info("job running for: " + job.getConfiguration().get(TOPIC_AND_HOUR) + ", url: " + job.getTrackingURL());
+      job.waitForCompletion(false);
+      if (!job.isSuccessful()) {
+        throw new RuntimeException("hadoop job failed.");
+      }
+    }
+
+    private void setSplitSizes(long targetSplitSize) {
```

Comment:
By setting the split size you are controlling the number of map tasks that get launched. You may want to add that to the javadocs for this method so that it is clear this code path determines the # of map tasks that get launched.

Also, is there any other way to control the number of map tasks that get launched? What if we want to throttle the hourly compaction job because it is taking up too many resources.